### DISCUSSION TITLE
Are there any examples of sending realtime data to Open MCT?

### DISCUSSION THREAD

**Discussion by jvigliotta** [nasa/openmct#4776] [Community Q&A] [answered]
_We've received a question similar to below, we will answer it here as it may benefit others as well._

We’re trying to use Open MCT to keep track of our telemetry but we are having problems in sending our data to the app. Do you have any examples of sending data to the app via tcp on a specific port and an example receiving it with a realtime plugin?

**Comment by akhenry** [comment, marked as answer]
Open MCT is a browser-resident JavaScript application for mission situational awareness. As such it cannot directly ingest telemetry data using regular TCP sockets as browsers don't support this.

Open MCT is designed to integrate with a separate telemetry and commanding system which would perform decommutation and processing of telemetry before making it available to Open MCT.  A number of different telemetry and commanding systems are support via plugins, including [Yamcs](https://yamcs.org/), [AIT](https://github.com/NASA-AMMOS/AIT-GUI), [AMPCS](https://github.com/NASA-AMMOS/MCWS) (coming soon), and others.
